Wsp/formplayer submit

wpride opened the pull request:
Switch to submitting forms from Formplayer rather than from the browser

PRed into wsp/django-auth, hold off on merge until that's merged

cross https://github.com/dimagi/formplayer/pull/52

biyeun committed:
add decorator for blazy

biyeun committed:
add block for full-page backgrounds

biyeun committed:
set up structure for new user registration view + form

biyeun committed:
add styles for backgrounds (todo documentation for full-page backgrounds)

biyeun committed:
add styling for step-through forms

biyeun committed:
add form step number to crispy

biyeun committed:
updates to new user registration form

biyeun committed:
updates to new user registration page (styling looks great, now need validation + click thru)

biyeun committed:
add knockoutValidation to bower and update decorators for use

biyeun committed:
update form structure

biyeun committed:
add RMI view to process registration form async requests

biyeun committed:
use project_name or hr_name from request

biyeun committed:
add some fancy to form steps

biyeun committed:
add validation message crispy forms element

biyeun committed:
finalize functionality for reg form

biyeun committed:
add password ko validator

biyeun committed:
update registration styles

biyeun committed:
no inline js in compress blocks

biyeun committed:
replace console.log with throw

biyeun committed:
lint

kaapstorm committed:
Notify when exception occurs in JSON response

kaapstorm committed:
Test notification

kaapstorm committed:
Check non-ASCII characters

biyeun committed:
show loading indicator for slow connections

biyeun committed:
make pageload a little smoother looking

biyeun committed:
remove unnecessary callback

biyeun committed:
update confirm text

biyeun committed:
change endpoint url

millerdev committed:
Update couch attachments view for external_blobs

biyeun committed:
base.html is a bad place for a 100% height and width dark background

biyeun committed:
attempts to fix weird django compressor issues

snopoke committed:
app index info

snopoke committed:
remove AppPillow from user tests

snopoke committed:
exctract function

snopoke committed:
app db kafka change publishing pillow

snopoke committed:
app pillow test

snopoke committed:
app to ES kafka pillow

benrudolph commented on corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js:
```diff
@@ -79,36 +79,10 @@ FormplayerFrontend.reqres.setHandler('startForm', function (data) {
+        FormplayerFrontend.request("clearForm");
+        // TODO form linking
+        FormplayerFrontend.trigger("apps:list");
+        showSuccess(gettext("Form successfully saved"), $("#cloudcare-notifications"), 2500);
```
so this shows success regardless of whether the form submission was successful?

wpride replied:
fixed, thanks.

snopoke committed:
test fixup

snopoke committed:
process all docs if no filter supplied

snopoke committed:
ignore empty chunks

snopoke committed:
replace default case reindexer with bulk resumable reindexer

snopoke committed:
use CASE_INDEX_INFO constant + import cleanup

snopoke committed:
fix typo

snopoke committed:
add AppDbChangeFeedPillow to all pillow meta

Commit:
ignore KEEP_UNTIL__ file

proteusvacuum committed:
Allow changing filters

dannyroberts committed:
include all xforms doc types in fast reindex

biyeun committed:
setup analytics hooks

millerdev committed:
Merge pull request #12320 from dimagi/sk/doc-processor-fixes

doc processor fixes

sravfeyn committed:
Merge pull request #12328 from dimagi/fr/compress-error

Remove css compress

sravfeyn committed:
resolve conflicts; add docs

sravfeyn committed:
Merge pull request #12329 from dimagi/fr/pillow-retry-styling

Add styling to pillow retry page

sravfeyn committed:
merge remote/master

sravfeyn committed:
lint

sravfeyn committed:
Merge pull request #12331 from dimagi/br/git-ignore-keep-until

ignore KEEP_UNTIL__ file

sravfeyn committed:
Merge pull request #12332 from dimagi/fr/cached-filters

Allow changing filters

snopoke committed:
add app to reindexer

sravfeyn committed:
check domain exists before getting attr

snopoke committed:
[ci skip] add todo

snopoke committed:
cleanup property after tests

snopoke committed:
cleaup before test case

snopoke committed:
report case bulk reindexer

sravfeyn committed:
pass real domain in get_test_payload

snopoke committed:
refactor paginate view into generic pagination function

sravfeyn committed:
kwarg -> arg

proteusvacuum committed:
Merge pull request #12333 from dimagi/fix-xform-fast-reindex

include all xforms doc types in fast reindex

millerdev committed:
Merge pull request #12316 from dimagi/sk/kafka-app-pillow

apps to ES kafka pillow + reindexer

millerdev committed:
Merge pull request #12335 from dimagi/sk/report-case-reindexer

report case bulk reindexer

millerdev committed:
Merge pull request #12082 from dimagi/2-session-api

2 session api

snopoke committed:
resumable function iterator

benrudolph committed:
Merge pull request #12323 from dimagi/sk/sql-reindexer-tweak

query db shards individually during reindexing

snopoke committed:
create ArgsProvider class to make contract clear

dannyroberts committed:
include all xforms doc types in v2 fast reindex

calellowitz committed:
adding analytics to hq notifications

snopoke committed:
add resume support

snopoke committed:
refactor ResumableDocsByTypeIterator to use ResumableFunctionIterator

snopoke committed:
fix tests

The need for this update is that previously we were only storing
'startkey' and 'endkey' in the resume state. Now we also store
'startkey_docid' and 'skip'.

snopoke committed:
fix rebase

proteusvacuum committed:
Remove appstore api

Commit:
a few fixes for formplayer

Commit:
remove dup line

calellowitz committed:
fixing bad template tags

millerdev committed:
Merge pull request #12338 from dimagi/ce/notification-analytics

adding analytics to hq notifications

calellowitz committed:
Merge pull request #12052 from dimagi/release-notifications

release notifications

NoahCarnahan committed:
Merge pull request #12228 from dimagi/icds-ls-static-ucrs

ICDS: Add Static Reports for LS App

calellowitz committed:
fixing date issues with gir

rickypchen committed:
added tooltip to clarify table numbers

Commit:
Make logic clearer on apply_rule return value

biyeun committed:
make sure pre-filled email works

Commit:
Allow case close rules to continue if parent case is not found

Commit:
Add ability to unretire user again

Commit:
add warning to user page

Commit:
[ci skip] dimagimon

Commit:
[ci skip] :facepunch:

dannyroberts committed:
Merge branch 'master' into fix-xform-fast-reindex-v2

biyeun committed:
create simple ab test variable generator

biyeun committed:
implement server-side ab tests

biyeun committed:
update prelogin submodule

biyeun committed:
url encode passed email string

biyeun committed:
improve email validation to handle a couple edge cases

Commit:
Properly handle name field, locations, and parents on import

millerdev committed:
Improve inventory ssh and mosh scripts

- Informative error messages.
- Pass extra arguments through to ssh/mosh. So you can now do `./scripts/inventory/mosh staging proxy -- tmux` to start a tmux session on the staging proxy.

sravfeyn committed:
Merge pull request #12348 from dimagi/br/fix-case-importer

Properly handle name field, locations, and parents on import

snopoke committed:
Merge pull request #12343 from dimagi/gc/case-close-update

Make logic clearer on apply_rule return value

snopoke committed:
Merge pull request #12337 from dimagi/fix-xform-fast-reindex-v2

include all xforms doc types in v2 fast reindex

snopoke committed:
no need to use .keys

snopoke committed:
remove reduntant if

snopoke committed:
don't yield changes for deleted

snopoke committed:
remove deleted docs from indexes

snopoke committed:
don't pass couchdb with every doc

snopoke committed:
extract progress logging to class

snopoke committed:
extract DocumentProvider class

snopoke committed:
refactor into DB agnostic document processors

snopoke committed:
fix test

snopoke committed:
rename module

snopoke committed:
pep8

kaapstorm committed:
Support Unicode messages

snopoke committed:
rename test module

snopoke committed:
remove unique_key from doc processor

snopoke committed:
fix reindexer

snopoke committed:
make document provider abstract

snopoke committed:
pass list of doc types directly to doc provider and logger

snopoke committed:
make reindexer DB agnostic

snopoke committed:
fix doc_types in ReportXFormReindexer

mkangia committed:
Handle Exception in case of download request for an invalid form

snopoke committed:
split module up

kaapstorm committed:
"and" in the wrong place

(Django template language doesn't allow brackets in "if"s, so nesting instead.)

millerdev committed:
Merge pull request #12342 from dimagi/ce/gir-bugs

fixing date issues with gir

millerdev committed:
Merge pull request #12350 from dimagi/sk/pr-feedback

pr feedback

snopoke committed:
sql doc processor classes

millerdev committed:
Merge pull request #12224 from dimagi/nh/notify_json_exc

Notify when exception occurs in JSON response

sravfeyn committed:
Merge pull request #12352 from dimagi/mk-error-form-download

Handle Exception in case of download request for an invalid form

orangejenny committed:
Merge pull request #12353 from dimagi/nh/229993/hide_usercase_adv

Don't show the "User Case Management" tab in Advanced Modules.

snopoke committed:
tests for resumable sql iterators

snopoke committed:
update sql form reindexer

snopoke committed:
fix logging

snopoke committed:
don't return None from doc transform

snopoke committed:
don't return None from doc transform

snopoke committed:
convert sql case reindexer

snopoke committed:
convert ledger v2 reindexer

snopoke committed:
lint

NoahCarnahan committed:
Merge pull request #12345 from dimagi/gc/case-close-fix

Allow case close rules to continue if parent case is not found

snopoke committed:
Merge pull request #12321 from dimagi/sk/fix-case-reindexer

replace default case reindexer with new one

snopoke committed:
include all doc types in reindexer

snopoke committed:
fix couch case reindexer

kaapstorm committed:
Rename constants to match Novartis demo app

kaapstorm committed:
Pass event case, not event_type

calellowitz committed:
waiting for document ready

benrudolph committed:
Merge pull request #12325 from dimagi/rc/ppdfixfirstmonthinactive

fixed bug - calculating first month of inactive users

benrudolph committed:
Merge pull request #12340 from dimagi/fr/remove-appstore-api

Remove appstore api

snopoke committed:
[ci skip] rename class and docs

NoahCarnahan committed:
Quote column name

benrudolph committed:
Merge pull request #12165 from dimagi/new-signup

New signup form w/ a/b test

kaapstorm committed:
Differentiate between domain name and Domain instance

dannyroberts committed:
Merge pull request #12349 from dimagi/dm/ssh-mosh-scripts

Improve inventory ssh and mosh scripts

dannyroberts committed:
Merge pull request #12351 from dimagi/sk/no-reindex-deleted-forms

don't yield changes for deleted

Commit:
Remove domain arg

Commit:
remove missing arg

snopoke committed:
fix imports

snopoke committed:
Merge branch 'master' into sk/pagination

snopoke committed:
Merge branch 'master' into sk/pagination

Commit:
delete_all_users command

calellowitz committed:
fixing js test

Commit:
update subs

rickypchen committed:
modified language to clarify table cells in project health dashboard

rickypchen committed:
removed tooltip function

orangejenny committed:
don't use javascript for IE warning

snopoke committed:
use function to get doc_type_map

snopoke committed:
fix test mechanism to discard state

snopoke committed:
remove slug arg from super call

dannyroberts committed:
Merge pull request #12354 from dimagi/sk/reindexer-doc-types

include all doc types in reindexer

snopoke committed:
Merge branch 'master' into sk/pagination

proteusvacuum committed:
Merge pull request #12355 from dimagi/ce/notification-analytics-2

waiting for document ready

orangejenny committed:
Merge pull request #12358 from dimagi/nh/study_data

Fix Study Data report for novartis-demo domain

biyeun committed:
Merge pull request #12362 from dimagi/ie-warning

Don't use javascript for IE warning

Commit:
fix up migration

calellowitz committed:
adding build profile options to mm download

calellowitz committed:
fixing formatting issue

rickypchen committed:
refactored to make html tags readble with tooltip titles

Commit:
fix travis build

esoergel committed:
Merge pull request #12368 from dimagi/br/fix-trav-build

fix travis build

Commit:
remove .py

Commit:
Merge pull request #12357 from dimagi/nc/expand-none-bug

Fix bug in UCR "expanded columns"

millerdev committed:
update staging.yaml

[ci skip]

millerdev committed:
Merge pull request #12364 from dimagi/ce/mm-download

Ce/mm download

Commit:
Remove report build tours

sravfeyn committed:
update dimagi-utils submodule

snopoke committed:
allow specifying chunksize via the command line

snopoke committed:
combine couch and sql case pillows

snopoke committed:
simplify tests

snopoke committed:
remove CasePillow from test

snopoke committed:
merge checkpoints for couch and sql case pillows

snopoke committed:
merge couch and sql form pillows

snopoke committed:
update tests

snopoke committed:
publish changes for deleted forms

snopoke committed:
reset resumable reindexers in tests

benrudolph committed:
Merge pull request #12373 from dimagi/dsr-stub

update dimagi-utils submodule

proteusvacuum committed:
Merge pull request #12360 from dimagi/br/update-subs

update subs

Commit:
Changed filter in App Status report

snopoke committed:
better handle errors in transormation

benrudolph committed:
Merge pull request #12376 from dimagi/sk/reset-reindexer-tests

reset resumable reindexers in tests

Commit:
small fixes

NoahCarnahan committed:
add senegal targets

snopoke committed:
[ci skip] more detailed docstring

snopoke committed:
copy doc_types list

snopoke committed:
rename function

Commit:
Merge pull request #12379 from dimagi/nc/abt-target-nums

Abt target numbers

snopoke committed:
add not about not supporting empty doc type list

Commit:
Remove message log link from messaging tab

orangejenny committed:
Set vcm flag on app creation only

benrudolph committed:
Merge pull request #12377 from lwyszomi/231760-case

Changed filter in App Status report

dannyroberts committed:
update staging.yaml

[ci skip]

orangejenny committed:
Add vellum_case_management flag to ES

orangejenny committed:
lint

NoahCarnahan committed:
Update strings

calellowitz committed:
tracking report type in email analytics

proteusvacuum committed:
Merge pull request #12370 from dimagi/dm/xform-blobs-reindex

Update couch attachments view for external_blobs

orangejenny committed:
vellum:f4430393abeac02b348586f0e53e599a905f70d1

esoergel committed:
Update deploy submodule

esoergel committed:
Merge pull request #12361 from dimagi/rc/ppdtooltip

Added tooltip to explain data in project performance dashboard

benrudolph committed:
Merge pull request #12388 from dimagi/update-deployment

Update deploy submodule

millerdev committed:
Merge pull request #12386 from dimagi/ce/email-report-types

tracking report type in email analytics

millerdev committed:
Merge pull request #12381 from dimagi/set-vcm-on-create

Set vellum case management flag on app creation only

millerdev committed:
Merge pull request #12326 from dimagi/sk/pagination

pagination

Commit:
Merge pull request #12385 from dimagi/nc/custom-expression-update

Minor updates to custom UCR expression

emord committed:
Merge pull request #12382 from dimagi/vcm-in-es

Add vellum_case_management flag to ES

Commit:
domain -> self.domain

esoergel committed:
Merge pull request #12389 from dimagi/br/fix-restore-commtrack-id

domain -> self.domain for commtrack restore

snopoke committed:
Merge pull request #12336 from dimagi/dsr-repeatertest

pass real domain to get_test_payload

kaapstorm committed:
Merge pull request #12341 from dimagi/br/flag-for-edit-forms

A few fixes for formplayer edit submissions

sravfeyn committed:
update diamgi-utils submodule

sravfeyn committed:
help message

czue committed:
Merge pull request #12258 from lwyszomi/icds_report_qa_2

Removed date filtering in Polpulation section for ASR report

snopoke committed:
retry sending to ES

snopoke committed:
topic list to single item + check for known topics

snopoke committed:
move app submission tracker pillow to it's own module

snopoke committed:
add reindexer for app submission tracker pillow

snopoke committed:
remove deleted doc type

czue committed:
Merge pull request #12394 from dimagi/sk/reindex-fixups

bulk reindexer fixups

NoahCarnahan committed:
Merge pull request #12372 from dimagi/br/remove-ucr-tours

Remove report build tours

NoahCarnahan committed:
Merge pull request #12380 from dimagi/gc/remove-message-log-link

Remove message log link from messaging tab

benrudolph committed:
Merge pull request #12391 from dimagi/dsr-stub

update diamgi-utils submodule

benrudolph committed:
Merge pull request #12392 from dimagi/dsr-help

help message

benrudolph committed:
Merge pull request #12375 from dimagi/sk/merge-couch-sql-pillows

merge couch and sql form/case kafka pillows

proteusvacuum committed:
Merge pull request #12346 from dimagi/br/ui-restore-user

Add ability to unretire user again

czue committed:
Merge pull request #12395 from dimagi/sk/app-submission-tracker-reindex

app submission tracker reindexer

Commit:
Merge branch 'master' into wsp/formplayer-submit

Commit:
subrepo update

Commit:
show error message if submit fails

dimagimon commented on corehq/apps/registration/static/registration/js/new_user.ko.js:
```diff
@@ -0,0 +1,310 @@
+                        }
+                    },
+                    message: django.gettext("There is already a user with this email."),
+                }
```
Missing trailing comma. (comma-dangle)

dimagimon commented on corehq/apps/app_manager/views/formdesigner.py:
```diff
@@ -115,6 +116,10 @@ def _form_too_large(app, form):
         ])
 
+    if tours.VELLUM_CASE_MANAGEMENT.is_enabled(request.user) \
+        and app.vellum_case_management and form.requires_case():
```
E125 continuation line with same indent as next logical line

dimagimon commented on corehq/pillows/xform.py:
```diff
@@ -2,30 +2,31 @@
-from pillowtop.reindexer.change_providers.couch import CouchViewChangeProvider
-from pillowtop.reindexer.reindexer import get_default_reindexer_for_elastic_pillow, \
-    ElasticPillowReindexer, ResumableElasticPillowReindexer
+from pillowtop.reindexer.reindexer import ElasticPillowReindexer, ResumableBulkElasticPillowReindexer
```
F401 'ElasticPillowReindexer' imported but unused

Commit:
Merge branch 'master' into wsp/formplayer-submit

benrudolph commented:
i think the branch you're PRing into needs a master merge

wpride commented:
Will recreate into master once that's merged

On Mon, Jul 11, 2016 at 2:58 PM, Ben <notifications@github.com> wrote:

> i think the branch you're PRing into needs a master merge
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/12308#issuecomment-231830825>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACL9SEZLWQJyk4unGf8ZglebDu2obxSjks5qUpJGgaJpZM4JDf6X>
> .
>

wpride commented:
close for PR into master https://github.com/dimagi/commcare-hq/pull/12402
